pages/popgen.py

```diff
@@ -0,0 +1,95 @@
+import holoviews as hv
+import numpy as np
+import panel as pn
+from bokeh import events
+from bokeh.models import ColumnDataSource
+from bokeh.models import CustomJS
+from holoviews import opts
+
+
+def make_one_way_stats(ts, windows_trees, windows_size, span_normalise, mode):
+    if windows_trees:
+        windows = "trees"
+        x_axis = ts.breakpoints(as_array=True)[:-1]
+    else:
+        windows = np.concatenate(
+            (np.arange(0, ts.sequence_length, windows_size), [ts.sequence_length])
+        )
+        x_axis = windows[:-1]
+    diversity = ts.diversity(span_normalise=span_normalise, windows=windows, mode=mode)
+    tajimas = ts.Tajimas_D(windows=windows, mode=mode)
+    diversity_curve = hv.Curve(
+        (x_axis, diversity), "Genomic position", "Diversity"
+    ).opts(interpolation="steps-post")
+    tajimas_curve = hv.Curve((x_axis, tajimas), "Genomic position", "Tajimas D").opts(
+        interpolation="steps-post"
+    )
+
+    def set_callbacks(fig, index, data):
+        source = ColumnDataSource(dict(index=index, x=data))
+
+        def hook(hv_plot, element):
+            fig = hv_plot.handles["plot"]
+            callback = CustomJS(
```

Comment:
What's this JS code for?

Reply:
This JS restricts panning and zooming to be horizontal. After any movement it adjusts the y axis range to be the extent of the visible data. You can see it working in the video at the top of the PR. Without this the mousewheel zooms in 2D for a frustrating experience!

The API it is using (`cb_obj`) is documented here: https://docs.bokeh.org/en/2.4.1/docs/user_guide/interaction/callbacks.html#customjs-for-user-interaction-events

Reply:
I see, sounds good. We'll probably want to do this across a few different plots then, so how about we pull it out for now and create an issue to track?